Makefile.build_config.in

```diff
@@ -70,6 +70,33 @@ MKDLL_EXP=@mkdll_exp@
 
 DEFAULT_BUILD_TARGET = @default_build_target@
 
+# Flags to use when compiling and linking OCaml programs
+
+# The names of the variables below take the form XXX_YYY_ZZZ where
+#
+# XXX is one of OC, OCAML or the name of a module. The OC prefix
+# is for private variables (i.e. reseverd by the compiler's build system),
+# the OCAML prefix is used for variables user can define to add their
+# own flags and the module-name prefix is for flags that apply only
+# to one module.
+#
+# YYY refers to the backend. At the moment, it can take the values
+# COMMON for the flags shared by all the backends, BYTECODE or NATIVE
+# (other backends may be added in the future).
+#
+# ZZZ is either CFLAGS (compile-tme flags) or LDFLAGS (link-time flags).
+# However, countrary to what is done for C compilers, the flags in the
+# CFLAGS category are not passed at link time, so if a flag is needed
+# at both stages, like e.e. -g, it should be added to both XXX_YYY_CFLAGS and
+# XXX_YYY_LDFLAGS.
+
+OC_COMMON_CFLAGS = -g -strict-sequence -principal -absname \
+  -w +a-4-9-40-41-42-44-45-48-66 -warn-error +a -bin-annot \
+  -safe-string -strict-formats
```

Comment:
FWIW we could use this opportunity to remove `-safe-string`, as it's only there for compatibility.